test/unit/solver/test_cuda_lu_decomposition.cpp

```diff
@@ -81,15 +42,15 @@ void testRandomMatrix(size_t n_grids)
   gpu_A.CopyToDevice();
   gpu_LU.first.CopyToDevice();
   gpu_LU.second.CopyToDevice();
-  gpu_lud.Decompose<double, GPUSparseMatrixPolicy>(gpu_A, gpu_LU.first, gpu_LU.second);
+  gpu_lud.Decompose<GPUSparseMatrixPolicy>(gpu_A, gpu_LU.first, gpu_LU.second);
   gpu_LU.first.CopyToHost();
   gpu_LU.second.CopyToHost();
-  check_results<double, GPUSparseMatrixPolicy>(
+  check_results<typename GPUSparseMatrixPolicy::value_type, GPUSparseMatrixPolicy>(
```

Comment:
It looks like the definition of `check_results` function has been deleted so I wonder how this execution here could work?

Reply:
that's now coming from the new header file